### PR TITLE
fix(kafka): Add back changes to centralize kafka config

### PR DESCRIPTION
**Pull request by dexter-mh-lee**
Changes made in this PR https://github.com/linkedin/datahub/pull/3083 was accidentally removed in this PR https://github.com/linkedin/datahub/pull/3015 

Adding back those changes so that schema registry is correctly configured for Glue schema registry. 

Resolves https://github.com/linkedin/datahub/issues/3373

## Checklist
- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

File: metadata-service/factories/src/main/java/com/linkedin/gms/factory/kafka/SimpleKafkaConsumerFactory.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+import org.springframework.context.annotation.Import;
+import org.springframework.kafka.config.ConcurrentKafkaListenerContainerFactory;
+import org.springframework.kafka.config.KafkaListenerContainerFactory;
+import org.springframework.kafka.core.DefaultKafkaConsumerFactory;
+
+
+@Slf4j
+@Configuration
+@EnableConfigurationProperties(KafkaProperties.class)
+@Import({KafkaSchemaRegistryFactory.class, AwsGlueSchemaRegistryFactory.class})
```
Comment (jjoyce0510) [inline metadata-service/factories/src/main/java/com/linkedin/gms/factory/kafka/SimpleKafkaConsumerFactory.java:24]: where are these factories used?

Comment (dexter-mh-lee) [reply metadata-service/factories/src/main/java/com/linkedin/gms/factory/kafka/SimpleKafkaConsumerFactory.java:24]: Good catch!

File: metadata-service/factories/src/main/java/com/linkedin/gms/factory/entity/RestliEntityClientFactory.java [hunk 2]
```diff
@@ -14,16 +14,16 @@
 @PropertySource(value = "classpath:/application.yml", factory = YamlPropertySourceFactory.class)
 public class RestliEntityClientFactory {
 
-  @Value("${DATAHUB_GMS_HOST:localhost}")
+  @Value("${datahub.gms.host}")
```
Comment (jjoyce0510) [inline metadata-service/factories/src/main/java/com/linkedin/gms/factory/entity/RestliEntityClientFactory.java:17]: awesome!

File: metadata-service/factories/src/main/resources/application.yml [hunk 3]
```diff
@@ -24,7 +24,13 @@ authentication:
...
   sessionTokenDurationMs: ${SESSION_TOKEN_DURATION_MS:86400000}
 
-
+datahub:
+  gms:
+    host: ${DATAHUB_GMS_HOST:${GMS_HOST:localhost}}
+    port: ${DATAHUB_GMS_PORT:${GMS_PORT:8080}}
+    useSSL: ${DATAHUB_GMS_USE_SSL:false}
+    sslContext:
+      protocol: ${DATAHUB_GMS_SSL_PROTOCOL:#{null}}
```
Comment (jjoyce0510) [inline metadata-service/factories/src/main/resources/application.yml:33]: Guessing these don't need to change?

**Review by jjoyce0510 — APPROVED** [review]
LGTM!

File: metadata-service/factories/src/main/resources/application.yml (hunk 3 above)
Comment (dexter-mh-lee) [reply metadata-service/factories/src/main/resources/application.yml:33]: Oh thought these were added by your PR. Adding these as well

**Review by shirshanka — APPROVED** [review]
LGTM!

**Pull request merged**